frontend/src/atoms/pipelineAtom.js

```diff
@@ -13,9 +13,10 @@ export const pipelineKey = (id, data) => {
 };
 
 export const pipelineFactory = (cachePath, pipeline = null) => {
-  const nanoid = customAlphabet("1234567890abcedfghijklmnopqrstuvwxyz", 12);
-  const newNanoid = nanoid();
-  const id = `pipeline-${newNanoid}`;
+  const id = generateId(
+    pipeline || { name: "pipeline" },
+    (pipeline && pipeline.name) || "pipeline",
+  );
```

Comment:
The previous code always would always generate `pipeline-{nanoid}`, so you should be able to just pass `"pipeline"` to `generateId` as the prefix.

Reply:
yeah. that's right. i am just going to pass 'pipeline' as a parameter now.